datanode/service/risk.go

```diff
@@ -81,7 +81,7 @@ func (r *Risk) ObserveMarginLevels(
 				return false
 			}
 			return (len(marketID) == 0 || marketID == acc.MarketID.String()) &&
-				(len(partyID) == 0 || partyID == acc.PartyID.String())
+				partyID == acc.PartyID.String()
```

Comment:
Is this a breaking change? before I could stream margin levels across all parties:
```
https://api.n06.testnet.vega.xyz/api/v2/stream/margin/levels
```
but now I can't? I suspect we also don't return a `InvalidArguement is required` error and just silently not match on anything in the stream.

Reply:
So the API documentation for this was:

```
// Request to subscribe to a stream of MarginLevels data matching the given party identifier
// Optionally, the list can be additionally filtered by market
message ObserveMarginLevelsRequest {
    // Party identifier, required field
    string party_id = 1 [(google.api.field_behavior) = REQUIRED];
    // Market identifier
    optional string market_id = 2;
}
```

I did check with @pscott31 and we agreed it would be more correct to make the implementation work as has been documented rather than the other way around. Happy to change it back and update the documentation on the API though.

Reply:
Addressed https://github.com/vegaprotocol/vega/pull/8070/commits/16a86a18b2d74fe341c6e3d4e19eb7563618cc95